### PR TITLE
WELD-1888, WELD-1891 - bean identifier index optimization

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by weld-tester** [comment]
Triggering build using a merge of 748bb1cbd83f82be24a7d53889c6d078b9ed678f on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 562 outcome was FAILURE using a merge of 748bb1cbd83f82be24a7d53889c6d078b9ed678f on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/562
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/562

**Comment by weld-tester** [comment]
Triggering build using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 563 is now running using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/563

**Comment by weld-tester** [comment]
Build 563 outcome was SUCCESS using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/563
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/563

**Comment by weld-tester** [comment]
Build 563 outcome was SUCCESS using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/563
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/563

**Comment by weld-tester** [comment]
Build 563 outcome was SUCCESS using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/563
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/563

**Comment by weld-tester** [comment]
Build 563 outcome was SUCCESS using a merge of 622e7166b216c734f8c8e979fae8ec0b4e77a3b5 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/563
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/563

File: impl/src/main/java/org/jboss/weld/serialization/BeanIdentifierIndex.java [hunk 1]
```diff
@@ -129,6 +132,8 @@ public int compare(BeanIdentifier o1, BeanIdentifier o2) {
         reverseIndex = builder.build();
 
         indexHash = Arrays.hashCode(index);
+
+        BootstrapLogger.LOG.beanIdentifierIndexBuilt(getDebugInfo());
```
Comment (jharting) [inline impl/src/main/java/org/jboss/weld/serialization/BeanIdentifierIndex.java]: We should add isDebugEnabled() check before building debug info

Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/serialization/BeanIdentifierIndex.java]: Good idea!

**Comment by jharting** [comment]
Also create a separate PR for 2.2

**Comment by weld-tester** [comment]
Triggering build using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 566 is now running using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by jharting** [comment]
Looks good to me. Go ahead with disabling this optimization by default for weld-servlet.

**Comment by weld-tester** [comment]
Build 566 outcome was SUCCESS using a merge of c07af35ea2acd8e7c9c81da687a0a21c1ad1ba56 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/566
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/566

**Comment by weld-tester** [comment]
Triggering build using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 579 is now running using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Build 579 outcome was SUCCESS using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Build 579 outcome was SUCCESS using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Build 579 outcome was SUCCESS using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Build 579 outcome was SUCCESS using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Build 579 outcome was SUCCESS using a merge of b192b9f7a554e9ebdee650fb69e8a346911b1136 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/579
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/579

**Comment by weld-tester** [comment]
Triggering build using a merge of 98d90709ef3f89c11e8bfb632eb055b86927e52c on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 592 is now running using a merge of 98d90709ef3f89c11e8bfb632eb055b86927e52c on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/592

**Comment by weld-tester** [comment]
Build 592 outcome was SUCCESS using a merge of 98d90709ef3f89c11e8bfb632eb055b86927e52c on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/592
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/592

**Comment by jharting** [comment]
Merged, thanks!

**Pull request closed without merging**